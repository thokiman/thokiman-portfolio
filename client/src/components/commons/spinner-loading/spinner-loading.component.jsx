import React from 'react';
import Loader from 'react-loader-spinner';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import useWindowDimensions from '../../hooks/window-dimensions/useWindowDimensions.component';
import { SpinnerContainer } from './spinner-loading.styles';
import { measureSizeSpinnerLoading } from './spinner-loading.utils';
export const SpinnerLoading = () => {
  const { viewWidth, viewHeight } = useWindowDimensions();
  const { widthSpinner, heightSpinner } = measureSizeSpinnerLoading(
    viewWidth,
    viewHeight
  );
  const whiteSmokeColor = '#e3e1e4';
  const loadingIcon = 'Watch';
  return (
    <SpinnerContainer>
      <Loader
        type={loadingIcon}
        color={whiteSmokeColor}
        height={heightSpinner}
        width={widthSpinner}
        visible
      />
    </SpinnerContainer>
  );
};

export default SpinnerLoading;
