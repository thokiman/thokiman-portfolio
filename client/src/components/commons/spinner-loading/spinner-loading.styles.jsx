import styled from 'styled-components';

const maxWidthSmartphone = 800;

export const SpinnerContainer = styled.div`
  height: 72vh;
  padding-left: 3vw;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: absolute;
    top: 25vh;
    padding: 0;
  }
`;

SpinnerContainer.displayName = 'SpinnerContainer';
