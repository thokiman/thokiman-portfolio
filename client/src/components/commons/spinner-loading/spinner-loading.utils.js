import {
  hasScreenSize1,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize4,
  hasScreenSize5,
} from '../../../utils/common.component.utils.js';

export const measureSizeSpinnerLoading = (viewWidth, viewHeight) => {
  let height;
  let width;

  if (hasScreenSize1(viewWidth)) {
    height = 300;
    width = 300;
  } else if (hasScreenSize2(viewWidth)) {
    height = 150;
    width = 150;
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    height = 140;
    width = 140;
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    height = 150;
    width = 150;
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    height = 130;
    width = 130;
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    height = 120;
    width = 120;
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    height = 100;
    width = 100;
  } else if (hasScreenSize5(viewWidth)) {
    height = 100;
    width = 100;
  } else {
    height = 200;
    width = 200;
  }
  return {
    widthSpinner: width,
    heightSpinner: height,
  };
};
