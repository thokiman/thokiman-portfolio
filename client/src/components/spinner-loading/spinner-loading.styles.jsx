import styled from "styled-components";

export const SpinnerContainer = styled.div`
  height: 72vh;
  padding-left: 3vw;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  @media only screen and (max-width: 800px) {
    height: 30vh;
    position: absolute;
    top: 20vh;
    padding: 0;
  }
`;

SpinnerContainer.displayName = "SpinnerContainer";
