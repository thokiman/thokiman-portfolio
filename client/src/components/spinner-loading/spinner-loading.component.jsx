import React from "react";

import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import { SpinnerContainer } from "./spinner-loading.styles";

const SpinnerLoading = () => {
  return (
    <SpinnerContainer>
      <Loader
        type="Watch"
        color="#e3e1e4"
        height={200}
        width={200}
        visible={true}
      />
    </SpinnerContainer>
  );
};

export default SpinnerLoading;
