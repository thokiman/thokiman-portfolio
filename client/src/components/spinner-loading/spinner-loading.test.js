import React from "react";
import { shallow } from "enzyme";
import { SpinnerLoading } from "./spinner-loading.component";

it("should render SpinnerLoading component", () => {
  expect(shallow(<SpinnerLoading />)).toMatchSnapshot();
});
