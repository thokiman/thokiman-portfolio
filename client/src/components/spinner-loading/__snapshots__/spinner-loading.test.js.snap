// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SpinnerLoading component 1`] = `ShallowWrapper {}`;
